server/src/main/java/com/broadcom/lsp/domain/cobol/databus/impl/CopybookRepositoryLRU.java

```diff
@@ -40,111 +39,55 @@
-    if (!isSort.get()) {
-      cpyRepo.sort(storableComparator.reversed().thenComparing(timeComparator));
-    }
-    isSort.set(true);
-  }
-
-  @Override
-  public Optional<CopybookStorable> getCopybookStorableFromCache(@NonNull long uuid) {
-    ArrayList<CopybookStorable> shallowCpy = (ArrayList<CopybookStorable>) cpyRepo.clone();
-    Optional<CopybookStorable> cpy =
-        shallowCpy.stream().filter(copy -> uuid == copy.getId()).findAny();
-    if (cpy.isPresent()) return cpy.map(SerializationUtils::clone);
-    return cpy;
+    cache = CacheBuilder.newBuilder()
+        .maximumSize(cacheMaxSize)
+        .expireAfterAccess(3, TimeUnit.HOURS)
+        .build();
   }
 
-  private Optional<CopybookStorable> getCopybookStorableInstance(@NonNull long uuid) {
-    return cpyRepo.stream().filter(copy -> uuid == copy.getId()).findAny();
+  /** For test purpose only */
+  CopybookRepositoryLRU(@Named("CACHE-MAX-SIZE") int cacheSize, Ticker ticker) {
+    cacheMaxSize = cacheSize;
+    cache = CacheBuilder.newBuilder()
+        .maximumSize(cacheMaxSize)
+        .expireAfterAccess(3, TimeUnit.HOURS)
+        .ticker(ticker)
+        .build();
   }
 
   @Override
-  public void setSort(boolean isSort) {
-    this.isSort.set(isSort);
+  public Optional<CopybookModel> getCopybookStorableFromCache(@NonNull String name) {
+    return Optional.ofNullable(cache.getIfPresent(name));
   }
 
   @Override
-  public synchronized void persist(@NonNull CopybookStorable deepCopy) {
-    cpyRepo.removeIf(CopybookStorable::isExpired);
-
-    if (!isStored(deepCopy.getId())) {
-      if (cpyRepo.size() < getCacheMaxSize()) {
-        cpyRepo.add(deepCopy);
-        return;
-      }
-      cpyRepo.remove(cpyRepo.size() - 1);
-      cpyRepo.add(deepCopy);
-    }
+  public synchronized void persist(@NonNull CopybookModel copybookModel) {
+    cache.put(copybookModel.getName(), copybookModel);
   }
 
   @Override
   public String logContent() {
-    StringBuilder chars = new StringBuilder();
-    cpyRepo.forEach(
-        l ->
-            chars
-                .append(System.getProperty("line.separator"))
-                .append(l)
-                .append(System.getProperty("line.separator")));
-    return chars.toString();
+    return cache.asMap().toString();
   }
 
   @Override
   public int size() {
-    return cpyRepo.size();
-  }
-
-  public Optional<CopybookStorable> topItem() {
-    return (cpyRepo.isEmpty()) ? Optional.empty() : Optional.of(cpyRepo.get(0));
-  }
-
-  public Optional<CopybookStorable> lastItem() {
-    return (cpyRepo.isEmpty()) ? Optional.empty() : Optional.of(cpyRepo.get(size() - 1));
-  }
-
-  @Override
-  public boolean isStored(@NonNull StringBuilder id) {
-    long uuid = CopybookRepository.calculateUUID(id);
-    return isStored(uuid);
-  }
-
-  @Override
-  public boolean isStored(@NonNull String id) {
-    long uuid = CopybookRepository.calculateUUID(id);
-    return isStored(uuid);
+    return (int)cache.size();
```

Comment:
I think is more safety to use ```Math.toIntExact(cache.size())```

Reply:
Yes, it will be safe. Didn't know about this one.